src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEModuleSymbol.cs

```diff
@@ -755,5 +757,27 @@ internal bool ShouldDecodeNullableAttributes(Symbol symbol)
 
             return false;
         }
+
+#nullable enable
+        internal bool GetCompilerFeatureRequiredUseSiteInfo(ref UseSiteInfo<AssemblySymbol> result)
+        {
+            bool highestPriorityUseSiteFound = false;
+            if (_lazyCachedUseSiteInfo.IsInitialized)
+            {
+                return MergeUseSiteInfo(ref result, _lazyCachedUseSiteInfo.ToUseSiteInfo(PrimaryDependency));
+            }
+
+            var decoder = new MetadataDecoder(this);
+            PEUtilities.DeriveUseSiteInfoFromCompilerFeatureRequiredAttributes(ref result, this, this, Token, CompilerFeatureRequiredFeatures.None, decoder);
+            highestPriorityUseSiteFound = result.DiagnosticInfo != null && IsHighestPriorityUseSiteError(result.DiagnosticInfo);
+
+            if (!highestPriorityUseSiteFound && ContainingAssembly is PEAssemblySymbol containingPEAssembly)
```

Comment:
>if (!highestPriorityUseSiteFound && ContainingAssembly is PEAssemblySymbol containingPEAssembly) [](http://example.com/codeflow?start=12&length=96)

This feels unnecessary complicated too. I would ask info from the assembly first and return it if it is there. Otherwise, I would return information attached to the module itself. Or I would do that in the opposite order. I really do not see a reason to deal with ```IsHighestPriorityUseSiteError``` because we are getting the same error (by id) for either symbol.